jwst/pipeline/calwebb_spec2.py

```diff
@@ -421,12 +421,10 @@ def _process_nirspec_masterbackground(self, data):
         """
         # First pass: just do the calibration to determine the correction
         # arrays.
-        pre_calibrated, ff_corrections = self.flat_field(
-            data, force_extended=True, return_corrections=True
-        )
-        pre_calibrated = self.pathloss(pre_calibratedforce_extended=True, return_corrections=True)
-        pre_calibrated = self.barshadow(pre_calibrated, force_extended=True, return_corrections=True)
-        pre_calibrated = self.photom(pre_calibrated, force_extended=True, return_corrections=True)
+        pre_calibrated = self.flat_field(data)
```

Comment:
Don't we want to still force all of these to be treated as extended sources, so that the computed corrections are appropriate for application to the master bkg data later on in the 2nd pass?

Reply:
Yes, and that will be coming with the Pathloss step.

Reply:
Ok, so you'll be adding that param back into the calling sequence for the other steps (pathloss, barshadow, photom)?